src/Land/Model/SoilHeatParameterizations.jl

```diff
@@ -184,23 +187,102 @@ end
 
 """
     volumetric_internal_energy_liq(
-        cp_l::FT,
         T::FT,
         T_ref::FT,
     ) where {FT}
+
 Compute the expression for the volumetric internal energy of liquid water.
-Here, cp_l is the volumetric heat capacity of liquid water.
 """
 function volumetric_internal_energy_liq(
     T::FT,
     param_set::AbstractParameterSet,
 ) where {FT}
-
     _T_ref = FT(T_0(param_set))
     _ρ_l = FT(ρ_cloud_liq(param_set))
     ρcp_l = FT(cp_l(param_set) * _ρ_l)
     ρe_int_l = ρcp_l * (T - _T_ref)
     return ρe_int_l
 end
 
+"""
+    function k_solid(
+        ν_ss_om::FT,
+        ν_ss_quartz::FT,
+        ν_ss_minerals::FT,
+        κ_quartz::FT,
+        κ_minerals::FT,
+        κ_om::FT,
+    ) where {FT}
+
+Computes the thermal conductivity of the solid material in soil.
+The `_ss_` subscript denotes that the volumetric fractions of the soil
+components are referred to the soil solid components, not including the pore
+space.
+"""
+function k_solid(
+    ν_ss_om::FT,
+    ν_ss_quartz::FT,
+    κ_quartz::FT,
+    κ_minerals::FT,
+    κ_om::FT,
+) where {FT}
+    return κ_om^ν_ss_om *
+           κ_quartz^ν_ss_quartz *
+           κ_minerals^(FT(1) - ν_ss_om - ν_ss_quartz)
+end
+
+
+"""
+    function ksat_frozen(
+        κ_solid::FT,
+        porosity::FT,
+        κ_ice::FT
+    ) where {FT}
+
+Computes the thermal conductivity for saturated frozen soil.
+"""
+function ksat_frozen(κ_solid::FT, porosity::FT, κ_ice::FT) where {FT}
+    return κ_solid^(FT(1.0) - porosity) * κ_ice^(porosity)
+end
+
+"""
+    function ksat_unfrozen(
+        κ_solid::FT,
+        porosity::FT,
+        κ_l::FT
+    ) where {FT}
+
+Computes the thermal conductivity for saturated unfrozen soil.
+"""
+function ksat_unfrozen(κ_solid::FT, porosity::FT, κ_l::FT) where {FT}
+    return κ_solid^(FT(1.0) - porosity) * κ_l^porosity
+end
+
+"""
+    function ρb_ss(porosity::FT, ρp::FT) where {FT}
+Computes the dry soil bulk density from the dry soil particle
+density.
+"""
+function ρb_ss(porosity::FT, ρp::FT) where {FT}
+    return (FT(1.0) - porosity) * ρp
+end
+
+"""
+    function k_dry(
+        κ_solid::FT,
+        porosity::FT,
+        κ_air::FT,
+        ρp::FT
+    ) where {FT}
+
+Computes the thermal conductivity of dry soil.
+"""
+function k_dry(κ_solid::FT, porosity::FT, κ_air::FT, ρp::FT) where {FT}
+    a = FT(0.053)
```

Comment:
Can we pull this constant out? It probably belongs as a model or free parameter.

Reply:
It's a good question, I'll think about it! Specifying these functions helped with model validation, but Im not sure down the road how much we will use them (vs. e.g. reading k_dry from a soil data set)

Reply:
OK, so we will be using a formula like this to compute k_dry from soil properties. I don't think we will want to tune them, though, as these functions have already been tuned by fitting soil data (and this soil data constrains them far more than our climate model would). 

Right now there are two other constants in the same category (used in the thermal conductivity model, but already tuned to soil data). These are stored in the soil model SoilParamFunctions - called a and b. I can add this one there as well.

Reply:
Waiting to make sure it all runs, but I think what I did is fine. I moved this constant out of this function:
https://github.com/CliMA/ClimateMachine.jl/blob/bdf33a47165f258a7eb78535941f4ca12e06306d/src/Land/Model/SoilHeatParameterizations.jl#L279

However, since k_dry was an attribute of soil_param_functions, and it needed that constant to evaluate, I couldnt pass the constant in via soil_param_functions. However, it seems clean enough for now to compute k_dry inside the source code (it only does it once). this just required making k_solid and rho_p (both things calculated from soil data sets; static profiles in space) attributes of soil_param_functions.